latex_access/polish_braille.py

```diff
@@ -0,0 +1,391 @@
+                self.lastnumber=-1
+                transden=self.translate(denominator[0],(rting[0]+denominator[2],rting[1]+len(transnum)+2))
+                translation=u"⠆"+transnum+u"⠳"+transden+u"⠰"
+        return (translation,denominator[1])
+
+    def bar(self, input, start,rting=()):
+        '''Handles bar/overline.
+
+        Returns toutple'''
+        arg=get_arg(input,start)
+        if len(arg[0])==1:
+            translation=":%s" % arg[0]
+            if rting!=(): self.rt.append((rting[1]+1,rting[0]+arg[2]))
+        else:
+            if rting==(): translation=":{%so" % self.translate(arg[0])
+            else: translation=":{%so" % self.translate(arg[0],(rting[0]+arg[2],rting[1]+2))
+        return (translation,arg[1])
+
+    def tag(self,input,start):
+        '''Translate  tags into Nemeth.
+
+        Returns a touple with translated string and index of
+        first char after end of tag.'''
+        arg=get_arg(input,start)
+        translation="  {"+arg[0]+"}"
+        return (translation,arg[1])
+
+    def log(self,input,start,rting=()):
+        '''Translate logs into polish braille.
+
+        Returns a touple with translated string and index of
+        first char after end of entire logarithm.'''
+        log=get_arg(input,start)
+        translation=u"⠫l"
+        if len(log[0]) < 1 or log[0][0] != "_": # \log by itself 
+            return (translation, log[2]) # ignore the supposed command 
+        
+        translation=u"⠌"
+        base=get_arg(input, log[1])
+        if base[0].isdigit():
+            for k in range(len(base[0])) : translation+=self.table["L"+base[0][k]]
+        else:
+            translation+= self.translate(base[0])
+        translation+=u"⠫l"
+        return (translation, base[1])
+
+    def pmod(self,input,start,rting=()):
+        '''Translates modules into polish braille.
+
+        Returns a touple, as above'''
+        arg=get_arg(input,start)
+        if arg[0].isdigit():
+            translation=u"⠈l"+self.translate(arg[0])+u"⠸"
+            if rting!=():
+                for j in range(len(arg[0])): self.rt.append((rting[1]+j,rting[0]+arg[2]+j))
+        else:
+            if rting==():
+                translation = u"⠈l"+self.translate(arg[0]) + u"⠸"
+            else:
+                translation = u"⠈l"+self.translate(arg[0],(rting[0]+arg[2],rting[1]+1)) + u"⠸"
+        return (translation,arg[1])
+
+    def letterSign (self,input,start):
+        '''Determines whether the letter sign is necessary. 
+
+        Returns a boolean.'''
+        lettersign = False # no lettersign yet
+        if input[start].lower () in 'aiouwz' and (self.lastnumber <= 0 or start != self.lastnumber) and ((start > 0 and len(input)-start>1 and input[start-1] == ' ' and input[start+1] == ' ') or (start == 0 and len(input)>1 and input[start+1] == ' ')): # special case no letter sign required, because these letter's are used as a separate words in polish
```

Comment:
I understand we don't have any particular requirements with regard to line lengths, but when this condition is in a single line it is completely unreadable. Would it be possible to split it up using parents?